### ISSUE TITLE
Move 2D/3D buttons away from the top

### ISSUE THREAD

**Issue by starry-abyss** [godotengine/godot-proposals#3153]
### Describe the project you are working on

A game with a lot of scene files and scripts.

### Describe the problem or limitation you are having in your project

![image](https://user-images.githubusercontent.com/11571820/129734397-5380d360-17e7-4aa3-a975-fc031334978a.png)

The 2D and 3D buttons add some confusion to my workflow, as they do 2 things at once - switching to scene editing mode and selecting how to preview the current scene (2D or 3D). I can accidentally switch a 2D scene to 3D view and have to click the other button then.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The scene preview mode is a scene option, and the most common scenario is that it's selected per scene and never changed afterwards. Hence we should put it near other scene-specific options.

![image](https://user-images.githubusercontent.com/11571820/129736287-e821b813-6542-4396-ba97-65a325113d0a.png)

Or it might even be hidden inside of the View menu.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![image](https://user-images.githubusercontent.com/11571820/129740565-47525071-e3ec-45a5-91fe-18e06651efb4.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

Add-ons can't mod this.

**Comment by starry-abyss** [comment]
Please take some time to explain what exactly you're confused about or ask questions instead of just using "confused" emoji. I have no idea what you mean by that. Thanks! :)

**Comment by pycbouh** [comment]
@starry-abyss I think confused reactions are usually used when respondents aren't sure why you want to change that or if it would be a good change, but aren't outright against it. It's probably hard to argue.

**Comment by dalexeev** [comment]
I understand the essence of the proposal and I am not strictly against it, but I am not sure that this proposal will be convenient for everyone, I think for some part of people the current workflow is preferable.

I think most of the confusion is caused by the third picture. Although it just simply means that the type (2D/3D) is remembered for each scene, as far as I understand. For most scenes this makes sense, but I can’t judge how convenient it is for mixed scenes, since I only work with 2D.

This is not the worst proposal, and probably even a good one, but additional arguments are needed. Also, in my opinion, the location of the "Switch to {N}D" button is not very good. We also need to decide whether we will leave support for the current option (not remember the scene type) and the ability to switch to it.

**Comment by starry-abyss** [comment]
The overall goal is to untangle the interface so simple things are simple. And if needed, untangle the editor behavior too.

Currently, when opening a script editor, it's shown as if it was a mode of the currently selected scene (inside the current tab). However, some experimentation reveals that it's global and has nothing to do with scene tabs.

Now if I want to go back to editing scenes, clicking on tabs is useless (I was going to address this in a separate PR later, while gathering opinions about this proposal in parallel), and 2D and 3D buttons imply a decision if I want 2D or 3D, while I just want to continue editing where I left and to not change anything.

> I think most of the confusion is caused by the third picture. Although it just simply means that the type (2D/3D) is remembered for each scene, as far as I understand.

It means that Script is global (regardless of this proposal), while 2D and 3D are not (regardless of this proposal), and they don't logically belong near the Script button. They would be global though if all scenes would be shown in 3D from the moment of selecting 3D - but that's not the case...

Whether 2D/3D is remembered or detected is really secondary here.

I think a huge portion of GDScript users actually edit code in VSCode, and that's why there is no issue for them going between Script and 2D/3D - they just don't use the internal editor at all.

**Comment by starry-abyss** [comment]
It's year 2024 now. Godot is a well-known game engine. Almost everyone knows that it can do 2D, 3D, GDScript and offers some community plugins before even downloading Godot. Also Godot has even more features than just those, which are hidden in other menus and panels, and I don't remember requests to put some features as buttons at top too.

At this point IMHO all these 4 buttons don't serve any special purpose rather than wasting space and making the engine look more childish than it is. I personally use all of these 4 features from time to time. But almost never click these buttons to use them. Let's move these buttons somewhere they belong more (in other menus and panels).

**Comment by AThousandShips** [comment]
Given that this hasn't gotten almost any support in over two years it doesn't look like this has wide demand, and I personally use these buttons regularly and don't think they should be moved behind further steps to activate

**Comment by romlok** [comment]
> Given that this hasn't gotten almost any support in over two years it doesn't look like this has wide demand, 

This probably doesn't see much demand/attention because it's just a papercut that doesn't make anything impossible to do, just makes it less efficient, especially for new users.
In terms of UX, putting these four buttons together makes little logical sense IMO. It's just something we all have gotten used to over time.

One big giveaway for me, that "2D" and "3D" should be scene-level controls instead of global-level, is that switching scenes can cause the view to change between 2D/3D, depending on the root node of the scene.

Additionally, it's sometimes desirable to see both 2D *and* 3D at the same time - eg. to make sure elements in a static 3D scene aren't hidden behind UI elements - which the current UI doesn't allow for AFAICT.

**Comment by Calinou** [comment]
> Additionally, it's sometimes desirable to see both 2D and 3D at the same time - eg. to make sure elements in a static 3D scene aren't hidden behind UI elements - which the current UI doesn't allow for AFAICT.

This is being tracked in https://github.com/godotengine/godot-proposals/issues/7513.
